apps/src/lib/node/ledger/shell/init_chain.rs

```diff
@@ -483,44 +621,251 @@ where
+            "\n\n{}\n{}\n{}\n\n",
+            separator,
+            "Report".bold().underline(),
+            separator
+        );
+        if self.errors.is_empty()
+            && self.panics.is_empty()
+            && self.warnings.is_empty()
+        {
+            println!(
+                "{}\n",
+                "Genesis files were dry-run successfully"
+                    .bright_green()
+                    .underline()
+            );
+            return;
+        }
+
+        if !self.warnings.is_empty() {
+            println!("{}\n\n", "Warnings".yellow().underline());
+            let warnings = Style::new().yellow();
+            for warning in &self.warnings {
+                println!("{}\n", warning.to_string().style(warnings));
+            }
+        }
+
+        if !self.errors.is_empty() {
+            println!("{}\n\n", "Errors".magenta().underline());
+            let errors = Style::new().magenta();
+            for error in &self.errors {
+                println!("{}\n", error.to_string().style(errors));
+            }
+        }
+
+        if !self.panics.is_empty() {
+            println!("{}\n\n", "Panics".bright_red().underline());
+            let panics = Style::new().bright_red();
+            for panic in &self.panics {
+                println!("{}\n", panic.to_string().style(panics));
+            }
+        }
+    }
+
+    /// Add a warning
+    fn warn(&mut self, warning: Warning) {
+        self.warnings.push(warning);
+    }
+
+    /// Categorize an error as normal or something that would panic.
+    fn register_err<E: Into<ErrorType>>(&mut self, err: E) {
+        match err.into() {
+            ErrorType::Runtime(e) => self.errors.push(e),
+            ErrorType::DryRun(e) => self.panics.push(e),
+        }
+    }
+
+    /// Categorize the error encountered and return a handle to allow
+    /// the code to specify how to proceed.
+    fn validate<T, E>(&mut self, res: std::result::Result<T, E>) -> Policy<T>
+    where
+        E: Into<ErrorType>,
+    {
+        match res {
+            Ok(data) => Policy {
+                result: Some(data),
+                dry_run: self.dry_run,
+            },
+            Err(e) => {
+                self.register_err(e);
+                Policy {
+                    result: None,
+                    dry_run: self.dry_run,
+                }
+            }
+        }
+    }
+
+    /// Check if any errors have been encountered
+    fn is_ok(&self) -> bool {
+        self.errors.is_empty() && self.panics.is_empty()
+    }
+
+    /// This should only be called after checking that `is_ok` returned false.
+    fn error_out(mut self) -> Result<()> {
+        if self.is_ok() {
+            return Ok(());
+        }
+        if !self.panics.is_empty() {
+            panic!(
+                "Namada ledger failed to initialize due to: {}",
+                self.panics.remove(0)
+            );
+        } else {
+            Err(self.errors.remove(0))
```

Comment:
don't we want a summary of all possible errors? so, something that cancatenates them into a single string, or a list (newline separated entries)

Reply:
The current design prints out all errors when dry-running. In "production", it errors out with the first error which was the original behavior.